### PR TITLE
Import Featurizers

### PR DESCRIPTION
**Pull request by yuslepukhin**
**Description**: This imports Featurizers library as optional external project so the sources
are pulled in only when necessary.

Disabled automl on CentOS since the source is not ready.

**Motivation and Context*
Previous import was in ORT code base. This imports a GH repository with the latest.

### CODE REVIEW

File: cmake/external/featurizers.cmake [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+# Only due to GIT_CONFIG
+# Uncoment UPDATE_COMMAND if you work locally
+# on the featurizers so cmake does not undo your changes.
+if (WIN32)
+    ExternalProject_Add(featurizers_lib
+            PREFIX ${featurizers_pref}
+            GIT_REPOSITORY ${featurizers_URL}
+            GIT_TAG ${ngraph_TAG}
+            # Need this to properly checkout crlf
```
Comment (snnn) [inline cmake/external/featurizers.cmake]: NGRAPH?

File: cmake/external/featurizers.cmake [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            SOURCE_DIR ${featurizers_ROOT}
+            # Location of CMakeLists.txt
+            SOURCE_SUBDIR src/Featurizers
+            BINARY_DIR ${featurizers_BINARY_DIR}
+            CMAKE_ARGS -DCMAKE_POSITION_INDEPENDENT_CODE=ON
+#            UPDATE_COMMAND ""
+            INSTALL_COMMAND ""
+        )
+endif()
+
```
Comment (snnn) [inline cmake/external/featurizers.cmake]: Just wondering, does the lib has any C++ types in its public interface?

File: cmake/external/featurizers.cmake (hunk 1 above)
Comment (yuslepukhin) [reply cmake/external/featurizers.cmake]: Good catch! May be that's the reason for some troubles.

---
In reply to: [357837997](https://github.com/microsoft/onnxruntime/pull/2643#discussion_r357837997) [](ancestors = 357837997)

File: cmake/external/featurizers.cmake (hunk 2 above)
Comment (yuslepukhin) [reply cmake/external/featurizers.cmake]: Well, they are not public  outisde ORT. ORT consumes this library.

---
In reply to: [357838315](https://github.com/microsoft/onnxruntime/pull/2643#discussion_r357838315) [](ancestors = 357838315)

**Pull request merged**